skyvern/schemas/workflows.py

```
@@ -547,6 +586,20 @@ class WorkflowDefinitionYAML(BaseModel):
 
+    @model_validator(mode="after")
+    def validate_unique_block_labels(cls, workflow: "WorkflowDefinitionYAML") -> "WorkflowDefinitionYAML":
+        labels = [block.label for block in workflow.blocks]
+        duplicates = [label for label in labels if labels.count(label) > 1]
+
+        if duplicates:
+            unique_duplicates = sorted(set(duplicates))
+            raise ValueError(
+                f"Block labels must be unique within a workflow. "
+                f"Found duplicate label(s): {', '.join(unique_duplicates)}"
+            )
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)